lib/grape/middleware/formatter.rb

```diff
@@ -171,7 +171,7 @@ def mime_array
         vendor_prefix_pattern = /vnd\.[^+]+\+/
 
         accept.scan(accept_into_mime_and_quality)
-              .sort_by { |_, quality_preference| -quality_preference.to_f }
+              .sort_by { |_, quality_preference| -(quality_preference || '1.0').to_f }
```

Comment:
Does it matter that it doesn't work with `quality_preference = ''`:

```ruby
3.1.2 :012 > quality_preference = ''
 => "" 
3.1.2 :013 > (quality_preference || '1.0').to_f
 => 0.0 
```

If it's really about `nil` (and not an explicitly passed in 0.0), then I would just do this to avoid parsing the 1.0 float every time.

```
quality_preference ? quality_preference.to_f : 1.0
```

Reply:
Yes, sounds good!